### PR TITLE
Refactor into internal package.

### PR DESCRIPTION
**Pull request by paddycarver**
An upcoming change is going to introduce a logging sink, which is going
to be a root logger that both our provider and SDK logs need to be
sub-loggers from, so they can inherit its filtering. This means that we
can no longer have the loggers siloed off from each other, or at least
need the sink to be accessible to both.

But that got me thinking: does it make _sense_ to have the loggers in
their own packages, or should the packages be boundaries around intended
audience? The number of "don't use this unless you're building an SDK"
comments in tflog make me sad, and make me wonder if it's better to just
say "tflog is for provider devs, tfsdklog is for SDK devs".

This PR is that. It refactors out the shared stuff between both loggers
into an internal/logging package, then uses that in tflog and tfsdklog.
It also moves things around so that only the functions we expect
provider devs to use are in the tflog package, only the functions we
expect SDK devs to use are in the tfsdklog package, and functions that
are only meant to be used by terraform-plugin-log (for testing, etc.)
are siloed off in the internal/testing package.

This feels much cleaner and less repetitive, and feels like a better
interface for provider devs and SDK devs alike.

### CODE REVIEW

File: internal/logging/log.go [hunk 1]
```diff
@@ -10,10 +10,22 @@ import (
 type loggerKey string
 
 const (
-	providerSpaceRootLoggerKey loggerKey = "provider"
+	// ProviderSpaceRootLoggerKey is the loggerKey that will hold the root
+	// logger for writing logs from within provider code.
+	ProviderSpaceRootLoggerKey loggerKey = "provider"
```
Comment (bflad) [inline internal/logging/log.go:15]: Nit: Any particular reason to include "Space" in the naming as compared to SDK? Any changes should also include the Get/Set functions. 👍

File: tfsdklog/options.go [hunk 2]
```diff
@@ -1,101 +1,59 @@
...
-	return func(l loggerOpts) loggerOpts {
-		l.name = name
+func WithLogName(name string) logging.Option {
+	return func(l logging.LoggerOpts) logging.LoggerOpts {
+		l.Name = name
 		return l
 	}
 }
 
+// WithStderrFromInit returns an option that tells the logger to write to the
```
Comment (bflad) [inline tfsdklog/options.go:20]: Nit: Not sure if we would want to keep this at the end of the file to reduce VCS "churn" and keep it alphabetical.

**Review by bflad — APPROVED** [review]
Love this 😍  Just left some very nitpicky things which are totally non-blocking and non-required.

File: internal/logging/log.go (hunk 1 above)
Comment (paddycarver) [reply internal/logging/log.go:15]: Not in particular, just some vague anxiety about it being confusing that I don't think panned out and I never went back to correct it. I'll fix it.

**Review by bflad — APPROVED** [review]
Still looks good to me 🚀

**Pull request merged**